packages/data-stores/src/mutations/use-domains-bulk-actions-mutation.ts

```diff
@@ -49,12 +55,26 @@ export function useDomainsBulkActionsMutation<
 					} );
 			}
 		},
+		onSuccess: ( data, variables ) => {
+			if ( variables.type !== 'set-auto-renew' ) {
+				return;
+			}
+
+			// Makes sure the success notice is shown once statuses have been updated
+			queryClient.invalidateQueries( { queryKey: getBulkDomainUpdateStatusQueryKey() } );
+
+			// Refreshes data for all the domains that were updated
+			variables.blogIds.forEach( ( blogId ) => {
+				queryClient.invalidateQueries( { queryKey: getSiteDomainsQueryKey( blogId ) } );
+				queryClient.invalidateQueries( { queryKey: getAllDomainsQueryKey( { no_wpcom: true } ) } );
```

Comment:
This invalidation request shouldn't be located in the `forEach()` method, as that call doesn't seem to be dependent on any of the arguments.